components/core/src/clp/streaming_archive/writer/Archive.cpp

```diff
@@ -593,16 +593,15 @@ void Archive::close_segment_and_persist_file_metadata(
 
     for (auto file : files) {
         file->mark_as_in_committed_segment();
+        // Files in this collection only hold metadata. Emplaced files have called method
+        // `File::append_to_segment()` which deallocates memory for timestamp,
+        // logtype, and variable fields.
+        m_file_metadata_for_global_update.emplace_back(file);
```

Comment:
This comment confused me a bit since this method doesn't seem to have anything to do with managing the lifecycle of the files (and the grammar is a bit odd). I think it would make more sense to add a comment above `m_file_metadata_for_global_update` in `Archive.hpp` indicating that the collection contains Files that have been written to the local archive and deallocated but whose metadata has not been persisted globally (rephrase it as appropriate).